### PR TITLE
Substract padding from list length when slicing

### LINKED ISSUE #3070: `smiles_from_seq` in `smiles_to_seq` has problem when `pad_len=0`.
## 🐛 Bug

In the [code](https://github.com/deepchem/deepchem/blob/10bf8db8f46b9870999bcc2d55deeff77574182c/deepchem/feat/molecule_featurizers/smiles_to_seq.py#L99) for `remove_pad` in `SmilesToSeq` class, first it extract the characters by removing additional padding process. So after that process it only has characters with the size of `max_len`. But when the case of `pad_len=0` it removes all the characters.
## To Reproduce

1. Featurizing smiles to sequences.
2. Convert sequences to smiles back.
3.

```
from rdkit import Chem
PAD_TOKEN = "<pad>"
OUT_OF_VOCAB_TOKEN = "<unk>"

char_to_idx = {'<unk>': 0, 
               '<pad>': 1,
               'C': 2, 
               'O': 3,}
max_len = 6
pad_len = 0
featurizer = SmilesToSeq(char_to_idx=char_to_idx, max_len=max_len, pad_len=pad_len)
example_smiles = ['CCCCO', 'CCC']
example_mols = [Chem.MolFromSmiles(smile) for smile in example_smiles]
featurized_smiles = featurizer.featurize(example_mols)

print(featurized_smiles)
print("length: ", len(featurized_smiles[1]))
print()

for seq in featurized_smiles:
    print("seq: ", seq)
    print(">>> To smiles: ")
    print(featurizer.smiles_from_seq(seq))
```
## Expected behavior
I expected the above code to return `CCCCO`, and `CCC` but it returns empty string.

## Environment

* OS: Colab
* Python version: Python 3.7.14
* DeepChem version: 2.6.1
* RDKit version (optional): 2022.03.5

* TensorFlow version (optional):
* PyTorch version (optional):
* Any other relevant information:

### PR DESCRIPTION
**Pull request by xerus**
# Pull Request Template

## Description

Avoid issues when pad_len is zero

Fix #3070

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by arunppsg — APPROVED** [review]
LGTM, thanks for the PR! Merging this in.
